### ISSUE TITLE
[Feature Request] Optional use of the AA download API

### ISSUE THREAD

**Issue by FlappyTaps** [calibrain/calibre-web-automated-book-downloader#11]
Annas Archive offers a download API for donators: https://annas-archive.org/dyn/api/fast_download.json

```
This API is intended as a stable JSON API for getting fast download files as a member.
A successful request will return status code 200 or 204, a `download_url` field and `account_fast_download_info`.
Bad responses use different status codes, a `download_url` set to `null`, and `error` field with string description.
Accepted query parameters:
- `md5` (required): the md5 string of the requested file.
- `key` (required): the secret key for your account (which must have membership).
- `path_index` (optional): Integer, 0 or larger, indicating the collection (if the file is present in more than one).
- `domain_index` (optional): Integer, 0 or larger, indicating the download server, e.g. 0='Fast Partner Server #1'.
These parameters correspond to the fast download page like this: /fast_download/{md5}/{path_index}/{domain_index}
Example: /dyn/api/fast_download.json?md5=d6e1dc51a50726f00ec438af21952a45&key=YOUR_SECRET_KEY
```

It would be great if we could use that instead of the cloudflare proxy to download books.

**Comment by IoSonoPiero** [comment]
You was faster than me to suggest that.
I completely agree.

**Comment by calibrain** [comment]
I don't have a key, but I can try to implement something and have someone test it.
I am pretty busy with end of year stuff, but will get to it eventually, or if someone feels courageous and want to send a PR I will review it 😄

**Comment by FlappyTaps** [comment]
I could potentially tackle that, but I'm usually not working with Python. Is there any guide on how to setup this locally in dev mode?

**Comment by calibrain** [comment]
Easiest way is to run the  `docker-compose.dev.yaml`, that one will build the docker image locally instead of sourcing it from the ghrc hub

Could you share an example of output of the API ?

**Comment by FlappyTaps** [comment]
I've created a working PR, not sure if that's how you'd structure it, but it works: https://github.com/calibrain/calibre-web-automated-book-downloader/pull/17

Here's an example API response:

```{
  "download_url": "https://directUrlToTheEbup.com/file.epub",
  "account_fast_download_info": {
    "downloads_left": 23,
    "downloads_per_day": 25,
    "recently_downloaded_md5s": [
      "md5hash1",
      "md5hash2"
    ]
  }
}```

**Comment by calibrain** [comment]
Thank you @FlappyTaps  !
Merged #17 and closing
Could someone try it and confirm if it works as expected ?
You can check the logs to see what is happening / which URL it used !

**Comment by IoSonoPiero** [comment]
It works!
Great job!
